### PR TITLE
ZCS-1212 Universal UI - Keyboard shortcuts preference visual update

### PR DESCRIPTION
**Pull request by tarangkhandelwal**
ZCS-1212 Universal UI - Keyboard shortcuts preference visual update

Changeset:
 * zm.css:
	* Updated shortcut layout to 2 columns from 3 columns.
	* Introduced ZmShortcutSeparator class as a horizontal separator support between each shortcut.
 * skin.properties: Styling updates as per the new UX mockup.

### CODE REVIEW

File: WebRoot/css/zm.css [hunk 1]
```diff
@@ -4398,18 +4398,21 @@ HTML>BODY .appt-selected .appt_allday_body
 
 .ZmShortcutList {
 	padding: 1rem;
-	-moz-columns: 3;
-	-webkit-columns: 3;
-	columns: 3;
+	-moz-columns: 2;
```
Comment (silentsakky) [inline WebRoot/css/zm.css]: can we create a property in skin.properties for this, so we will be able to check browser type and apply specific prefixed version, check DisableSelection for example

File: WebRoot/skins/_base/base4/skin.properties [hunk 2]
```diff
@@ -1155,6 +1155,15 @@ FormAddRemoveButton          = @FormButtonHeight@
 FormButtonStyle              = @cssShadow@:0px 0px 0px transparent; padding:5px; @ActiveCursor@ @PanelBorder@ @MediumRoundCorners@ background-color:@WidgetBgColor@; @ButtonText@
 
 ###################
+#   SHORTCUT LIST
+###################
+PrefShortcutLabel            = min-width: 50%; margin-bottom: 4px; @FontSize-big@ @Text@
+PrefShortcutKey              = line-height:3.667em;margin:0 0.667em; padding:6px 10px; @Text@ @SeparatorBorder@ @MediumRoundCorners@
+ShortcutListTitle            = padding:1rem 0; font-size:1.5rem; @Text@
+ShortcutList                 = margin: 0 20px 0 0; break-inside: avoid-column;
+ShortcutSeparator            = border-bottom: 1px dashed #CCCCCC; margin:6px 0;
```
Comment (silentsakky) [inline WebRoot/skins/_base/base4/skin.properties]: not sure but can we use some property instead of #CCC?

Comment (SatyamKumarOnline) [inline WebRoot/skins/_base/base4/skin.properties]: SeparatorColor can be used for #CCCCCC

File: WebRoot/css/zm.css (hunk 1 above)
Comment (tarangkhandelwal) [inline WebRoot/css/zm.css]: I am not very sure going that way. There doesn't seem to be a block for webkit , just GECKO, SAFAR & default I guess.

Comment (silentsakky) [inline WebRoot/css/zm.css]: I am not exactly sure but we can use any of below macros, we need to check this

// operating systems
			define(macros, "LINUX", isLinux);
			define(macros, "MACINTOSH", isMac);
			define(macros, "WINDOWS", isWindows);

			// browser variants
			define(macros, "CHROME", isChrome);
			define(macros, "CHROME_4_OR_HIGHER", isChrome4up);
			define(macros, "FIREFOX", isFirefox);
			define(macros, "FIREFOX_1_OR_HIGHER", isFirefox1up);
			define(macros, "FIREFOX_1_5_OR_HIGHER", isFirefox1_5up);
			define(macros, "FIREFOX_4_OR_HIGHER", isFirefox4up);
			define(macros, "GECKO", isGeckoBased);
			define(macros, "GECKO_1_8_OR_HIGHER", isGecko1_8up);
			define(macros, "GECKO_2_OR_HIGHER", isGecko2up);
			define(macros, "HOTJAVA", isHotJava);
			define(macros, "IPHONE", isIPhone);
			define(macros, "MOZILLA", isMozilla);
			define(macros, "MOZILLA_1_4_OR_HIGHER", isMozilla1_4up);
			define(macros, "MSIE", isIE);
			define(macros, "MSIE_3", isIE3);
			define(macros, "MSIE_4", isIE4);
			define(macros, "MSIE_4_OR_HIGHER", isIE4up);
			define(macros, "MSIE_5", isIE5);
			define(macros, "MSIE_5_OR_HIGHER", isIE5up);
			define(macros, "MSIE_5_5", isIE5_5);
			define(macros, "MSIE_5_5_OR_HIGHER", isIE5_5up);
			define(macros, "MSIE_6", isIE6);
			define(macros, "MSIE_6_OR_HIGHER", isIE6up);
			define(macros, "MSIE_LOWER_THAN_7", isIE && !isIE7up);
			define(macros, "MSIE_7", isIE7);
			define(macros, "MSIE_7_OR_HIGHER", isIE7up);
			define(macros, "MSIE_8", isIE8);
			define(macros, "MSIE_8_OR_HIGHER", isIE8up);
			define(macros, "MSIE_LOWER_THAN_9", isIE && !isIE9up);
			define(macros, "MSIE_9", isIE9);
			define(macros, "MSIE_9_OR_HIGHER", isIE9up);
			define(macros, "MSIE_LOWER_THAN_10", isIE && !isIE10up);
			define(macros, "MSIE_10", isIE10);
			define(macros, "MSIE_10_OR_HIGHER", isIE10up);
			define(macros, "MODERN_IE", isModernIE);
			define(macros, "NAVIGATOR", isNav);
			define(macros, "NAVIGATOR_4", isNav4);
			define(macros, "NAVIGATOR_6", isNav6);
			define(macros, "NAVIGATOR_6_OR_HIGHER", isNav6up);
			define(macros, "NAVIGATOR_7", isNav7);
			define(macros, "NAVIGATOR_COMPATIBLE", isCompatible);
			define(macros, "OPERA", isOpera);
			define(macros, "SAFARI", isSafari);
			define(macros, "SAFARI_2", isSafari2);
			define(macros, "SAFARI_2_OR_HIGHER", isSafari2up);
			define(macros, "SAFARI_3", isSafari3);
			define(macros, "SAFARI_5_OR_HIGHER", isSafari5up);
			define(macros, "TRIDENT", isTrident);
			define(macros, "WEBKIT", isWebKitBased);
			define(macros, "WEBTV", isWebTv);

Comment (tarangkhandelwal) [inline WebRoot/css/zm.css]: Updated here:
https://github.com/Zimbra/zm-web-client/pull/72/commits/660b72cce4446724fd53799ba2f0794d6eb4cf38

File: WebRoot/css/zm.css [hunk 3]
```diff
@@ -4431,7 +4431,7 @@ HTML>BODY .appt-selected .appt_allday_body
 
 .shortcutListMap DT {
 	display: inline-block;
-	text-align: right;
+	text-align: left;
```
Comment (SatyamKumarOnline) [inline WebRoot/css/zm.css]: It can be moved to properties file

Comment (tarangkhandelwal) [inline WebRoot/css/zm.css]: Yes we can move this one to be as a property, but just changed alignment here and there are almost 20+ style rules related to shortcut here that are not using properties. Ideally if we think these should be customisable, we should change all the rules in one go. 
Let me know your thoughts on this.

File: WebRoot/js/zimbraMail/prefs/view/ZmShortcutsPage.js [hunk 4]
```diff
@@ -107,7 +108,8 @@ function() {
 
 	var html = [], i = 0;
 	html[i++] = "<html><head>";
-	html[i++] = "<link href='" + appContextPath + "/css/zm.css' rel='stylesheet' type='text/css' />";
+	//adding skin query param in the css being loaded to load compiled stylesheet.
+	html[i++] = "<link href='" + appContextPath + "/css/zm.css?skin="+ window.skin.hints.name +"' rel='stylesheet' type='text/css' />";
```
Comment (SatyamKumarOnline) [inline WebRoot/js/zimbraMail/prefs/view/ZmShortcutsPage.js]: can we use getSkinHint API, here

File: WebRoot/skins/_base/base4/skin.properties [hunk 5]
```diff
@@ -1155,6 +1155,16 @@ FormAddRemoveButton          = @FormButtonHeight@
 FormButtonStyle              = @cssShadow@:0px 0px 0px transparent; padding:5px; @ActiveCursor@ @PanelBorder@ @MediumRoundCorners@ background-color:@WidgetBgColor@; @ButtonText@
 
 ###################
+#   SHORTCUT LIST
+###################
+PrefShortcutLabel            = min-width: 50%; margin-bottom: 4px; @FontSize-big@ @Text@
```
Comment (SatyamKumarOnline) [inline WebRoot/skins/_base/base4/skin.properties]: Some of the properties used here, in this block, have space after  : and ;
While others donot have.
Can we have uniform indentation for property and values

File: WebRoot/js/zimbraMail/prefs/view/ZmShortcutsPage.js (hunk 4 above)
Comment (tarangkhandelwal) [inline WebRoot/js/zimbraMail/prefs/view/ZmShortcutsPage.js]: sure, replacing it with appCtxt.getSkinHint("name"): ae27f22117be80e8f1be2d5f058ce51dafc87bcc

File: WebRoot/skins/_base/base4/skin.properties (hunk 5 above)
Comment (tarangkhandelwal) [inline WebRoot/skins/_base/base4/skin.properties]: Done here: 8674ff1272c171af3e9a1c1b9ccedc755363dbc9
Thanks.

File: WebRoot/skins/_base/base4/skin.properties [hunk 6]
```diff
@@ -1155,6 +1155,16 @@ FormAddRemoveButton          = @FormButtonHeight@
 FormButtonStyle              = @cssShadow@:0px 0px 0px transparent; padding:5px; @ActiveCursor@ @PanelBorder@ @MediumRoundCorners@ background-color:@WidgetBgColor@; @ButtonText@
 
 ###################
+#   SHORTCUT LIST
+###################
+PrefShortcutLabel            = min-width:50%; margin-bottom:4px; @FontSize-big@ @Text@
+PrefShortcutKey              = line-height:3.667em; margin:0 0.667em; padding:6px 10px; @Text@ @SeparatorBorder@ @MediumRoundCorners@
```
Comment (SatyamKumarOnline) [inline WebRoot/skins/_base/base4/skin.properties]: Prefer unitless numbers for line-height values.
https://developer.mozilla.org/en/docs/Web/CSS/line-height
https://css-tricks.com/almanac/properties/l/line-height/

File: WebRoot/skins/_base/base4/skin.properties [hunk 7]
```diff
@@ -1155,6 +1155,16 @@ FormAddRemoveButton          = @FormButtonHeight@
 FormButtonStyle              = @cssShadow@:0px 0px 0px transparent; padding:5px; @ActiveCursor@ @PanelBorder@ @MediumRoundCorners@ background-color:@WidgetBgColor@; @ButtonText@
 
 ###################
+#   SHORTCUT LIST
```
Comment (SatyamKumarOnline) [inline WebRoot/skins/_base/base4/skin.properties]: In this section, see different units like px, rem, em being used.
Any specific reason for this.
Curious about this.

File: WebRoot/css/zm.css (hunk 3 above)
Comment (SatyamKumarOnline) [inline WebRoot/css/zm.css]: Ok. Thanks for explanation.

File: WebRoot/skins/_base/base4/skin.properties (hunk 6 above)
Comment (tarangkhandelwal) [inline WebRoot/skins/_base/base4/skin.properties]: Unitless value not adding any value as per my understanding here (as per the link's useful when we have child elements). Used relative sizing based on parent's font size.

File: WebRoot/skins/_base/base4/skin.properties (hunk 7 above)
Comment (tarangkhandelwal) [inline WebRoot/skins/_base/base4/skin.properties]: It is not a recommendation to only use one unit. I chose to use value's that made more sense considering the impact. For ex: If we want to have fixed margin irrespective of zoom or parent font-size : use px. 
If we want to have breathing space's increased on zoom or with increase in parent font-size use em/rem.

Comment (SatyamKumarOnline) [inline WebRoot/skins/_base/base4/skin.properties]: ok. Thanks

**Pull request merged**